src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,212 @@
+#:include "common.fypp"
+module stdlib_stats_distribution_PRNG
+    use stdlib_kinds, only: int8, int16, int32, int64
+    implicit none
+    private
+    integer, parameter :: MAX_INT_BIT_SIZE = bit_size(1_int64)
+    integer(int64), save :: st(4), si = 614872703977525537_int64
```

Comment:
Should the states `st(4)` be not defined for each kind?

For example, if I need to generate random numbers of `int32` and `int64` in a same loop:
```fortran
do i = 1, n
 ...
 res_int32(i) = dist_rand(1_int32)
 ...
 res_int64(i) = dist_rand(1_int64)
 ...
enddo
```
 what would happen with the states?
IMO it is not a problem, because it seems to always use `int64` internally. But I though it would be worthwhile to discuss it.

Reply:
Yes, internally there is only one int64 random number generator. All other integer kinds like int8, int16 and int32 random numbers are generated by converting higher int64 integer bits through shift operation into specific kind. For instance, if we need an int32 random integer, we first generate an int64 random integer, only the upper 32 bits of the int64 integer is taken to give an int32 random integer.

So, st(4) is only for int64 kind. There is no need of st(4) for other kinds.